examples/stable_diffusion_xl/gm/modules/diffusionmodules/sigma_sampling.py

```diff
@@ -12,8 +12,9 @@ def __init__(self, p_mean=-1.2, p_std=1.2):
         self.p_mean = p_mean
         self.p_std = p_std
 
-    def construct(self, n_samples, rand=None):
-        log_sigma = self.p_mean + self.p_std * default(rand, ops.randn((n_samples,)))
+    @ms.jit  # ops.randn causes 'dynamic shape' error in PyNative mode
```

Comment:
maybe add a `#FIXME` tag here since it is a temporary bug caused by Mindspore

Reply:
I suppose under PyNative mode, there shouldn't be a `dynamic shape` error. If you set `n_samples` as the attribute of `EDMSampling`, and using `ops.randn((self.n_samples, ))`, will the same error persist?

Reply:
This seems odd to me too, but the error is thrown during backpropagation.
I haven't checked whether saving the variable as a class attribute would help, since it's not always possible (e.g. with `gm.modules.distributions.distributions.DiagonalGaussianDistribution.sample`).